### ISSUE TITLE
Azure data disk should provision storage account on on-demand

### ISSUE THREAD

**Issue by karataliu** [kubernetes/kubernetes#50883]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
The azure_dd plugin will always provision two instances of storage account with different type during init.

There could be cases when:
1. User may only want to use one certain type of storage account, then the other type of storage account would be not in use.
2. User chooses to use managed disk, both of the storage account would be not in use. 

There is a limit for storage account numbers per subscription, and the storage account should be created with a on-demand policy.

Log:
```
I0818 03:00:14.709321       1 azure_blobDiskController.go:354] azureDisk - BlobDiskController init process  will create new storageAccount:pvc2365976634001 type:Premium_LRS
I0818 03:00:14.709693       1 azure_blobDiskController.go:354] azureDisk - BlobDiskController init process  will create new storageAccount:pvc2365976634002 type:Standard_LRS
I0818 03:00:14.739106       1 azure_blobDiskController.go:602] azureDisk - Creating storage account pvc2365976634001 type Premium_LRS 
I0818 03:00:14.759665       1 azure_blobDiskController.go:602] azureDisk - Creating storage account pvc2365976634002 type Standard_LRS
```
Code:
https://github.com/kubernetes/kubernetes/blob/9e97b5249bdd105705bfa6a7c6da477e8bde57d1/pkg/cloudprovider/providers/azure/azure_blobDiskController.go#L335-L366

**What you expected to happen**:
The storage accounts be created on-demand.

**How to reproduce it (as minimally and precisely as possible)**:
Setup a kubernetes v1.7.2 cluster on Azure, create a dynamic pvc.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.0", GitCommit:"d3ada0119e776222f11ec7945e6d860061339aad", GitTreeState:"clean", BuildDate:"2017-06-29T23:15:59Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"7", GitVersion:"v1.7.2", GitCommit:"922a86cfcd65915a9b2f69f3f193b8907d741d9c", GitTreeState:"clean", BuildDate:"2017-07-21T08:08:00Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration**:
Azure

**Comment by khenidak** [comment]
Creating a storage account is a lengthy process (~ 1.5 mins), these 2 accounts are created upfront to ensure that PVCs will be quickly provisioned into existing storage accounts. Since ACS & acs-engine currently supports mixed (managed and blob) disks we won't be able to tell if they user will or won't use blob disks. Maybe once acs-* moves away from blob disks we can disable them. They have no impact on users since users get billed only for data inside storage accounts not storage accounts themselves

**Comment by andyzhangx** [comment]
for the "limit for storage account numbers per subscription" issue, I have another issue to track this: https://github.com/kubernetes/kubernetes/issues/52396, will fix it in v1.9 and cherry pick to v1.8, v1.7
